src/system/global_defs.h

```diff
@@ -152,50 +152,71 @@ if ( !(expr) ) \
 
```

Comment:
It would be better to move all new assert macros into asserts.hh and include asserts.hh from global_defs.h. That would be temporary solution until we finish system classes.